### PR TITLE
SISRP-25702 - CS Link errors logged without UID

### PR DESCRIPTION
**Pull request by dfkaye**
https://jira.berkeley.edu/browse/SISRP-25702

### CODE REVIEW

File: app/models/campus_solutions/student_resources.rb [hunk 1]
```diff
@@ -9,28 +9,29 @@ def initialize(options = {})
...
       empl_id = lookup_campus_solutions_id
-      link_ids = {
-        change_of_academic_plan_add: 'UC_CX_GT_CPPSTACK_ADD',
-        change_of_academic_plan_view: 'UC_CX_GT_CPPSTACK_VIEW',
-        emergency_loan_form_add: 'UC_CX_GT_FAEMRLAON_ADD',
-        emergency_loan_form_view: 'UC_CX_GT_FAEMRLAON_VIEW',
-        withdraw_from_semester_add: 'UC_CX_SRWITHDRL_ADD',
-        veterans_benefits_add: 'UC_CX_GT_SRVAONCE_ADD'
-      }
+      return {} unless !empl_id.blank?
```
Comment (johncrossman) [inline app/models/campus_solutions/student_resources.rb]: @dfkaye How about simplified:
```
return {} if empl_id.blank?
```

File: app/models/campus_solutions/student_resources.rb [hunk 2]
```diff
@@ -9,28 +9,29 @@ def initialize(options = {})
...
+
+      cs_links = {}
+
+      campus_solutions_link_settings = [
+        { feed_key: :change_of_academic_plan_add, cs_link_key: 'UC_CX_GT_CPPSTACK_ADD', cs_link_params: { } },
+        { feed_key: :change_of_academic_plan_view, cs_link_key: 'UC_CX_GT_CPPSTACK_VIEW', cs_link_params: { } },
+        { feed_key: :emergency_loan_form_add, cs_link_key: 'UC_CX_GT_FAEMRLAON_ADD', cs_link_params: { } },
+        { feed_key: :emergency_loan_form_view, cs_link_key: 'UC_CX_GT_FAEMRLAON_VIEW', cs_link_params: { } },
+        { feed_key: :withdraw_from_semester_add, cs_link_key: 'UC_CX_SRWITHDRL_ADD', cs_link_params: { } },
+        { feed_key: :veterans_benefits_add, cs_link_key: 'UC_CX_GT_SRVAONCE_ADD', cs_link_params: { } },
```
Comment (johncrossman) [inline app/models/campus_solutions/student_resources.rb]: @dfkaye why do we have `cs_link_params` if all entries have empty `{}`?

File: app/models/campus_solutions/student_resources.rb [hunk 3]
```diff
@@ -9,28 +9,29 @@ def initialize(options = {})
...
+      ]
 
-      link_ids.each do |key, url_id|
-        link = CampusSolutions::Link.new.get_url(url_id, { :EMPLID => empl_id }).try(:[], :link)
-        if link
-          resource_links[key.to_s.camelize(:lower).to_sym] = link
-        end
+      campus_solutions_link_settings.each do |setting|
+        link = fetch_link(setting[:cs_link_key], setting[:cs_link_params])
```
Comment (johncrossman) [inline app/models/campus_solutions/student_resources.rb]: @dfkaye if you do away with `cs_link_params` above then we can drop `setting[:cs_link_params]` here. I recommend we bring it back when we have a need.

**Comment by dfkaye** [comment]
@johncrossman your suggestions always make sense - they're in now...

**Comment by johncrossman** [comment]
Looks good, @dfkaye

**Pull request merged**